internal/app/app.go

```diff
@@ -0,0 +1,89 @@
+package app
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/metal-toolbox/conditionorc/internal/model"
+	"github.com/spf13/viper"
+
+	runtime "github.com/banzaicloud/logrus-runtime-formatter"
+	"github.com/sirupsen/logrus"
+)
+
+// Config holds configuration data when running mctl
+// App holds attributes for the mtl application
+type App struct {
+	// Viper loads configuration parameters.
+	v *viper.Viper
+	// The kind of application - server/orchestrator
+	AppKind model.AppKind
+	// Sync waitgroup to wait for running go routines on termination.
+	SyncWG *sync.WaitGroup
+	// Flasher configuration.
+	Config *Configuration
+	// TermCh is the channel to terminate the app based on a signal.
+	TermCh chan os.Signal
```

Comment:
This also feels like a weird export since its not limited to blocking reads for callers. Maybe consider returning a read-only ch in New or some other method?

Reply:
That makes sense, fixed in [83a08ce](https://github.com/metal-toolbox/conditionorc/pull/1/commits/83a08ce04ce5bde99c8c0a625770c2ba819c37c0)